internal/http/transports.go

```diff
@@ -183,3 +184,23 @@ func (s ConnSettings) NewRemoteTargetHTTPTransport(insecure bool) func() *http.T
 		return tr
 	}
 }
+
+// uaTransport - User-Agent  transport
+type uaTransport struct {
+	ua string
+	rt http.RoundTripper
+}
+
+func (u *uaTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req2 := req.Clone(req.Context())
+	req2.Header.Set("User-Agent", u.ua)
+	return u.rt.RoundTrip(req2)
+}
+
+// WithUserAgent wraps an existing transport with custom User-Agent
+func WithUserAgent(rt http.RoundTripper) http.RoundTripper {
```

Comment:
Honestly it feels slightly overengineered that this isn't just...
```suggestion
func WithUserAgent(rt http.RoundTripper, getUA func() string) http.RoundTripper {
```

That also makes it very obvious where the useragent is actually from, and not that it is pulled from a global var somewhere.

Simpler to understand and simpler to use, no need to register anything.

Reply:
yes I agree, the problem I was getting import cycle error due to the getUserAgent() living in package cmd

Reply:
What do you mean? You aren't importing more. You are just sending a `func() string`.

Reply:
ohh, not here. I meant the place where I would be using the WithUserAgent in for example in a different package openid in jwt.go

Reply:
Oh, with jwt... Well, you could just wrap it on the 2 places `openid.LookupConfig` is called.

Reply:
okay, this sounds good. I will update the code